### PR TITLE
Add command persist.

### PR DESCRIPTION
**Pull request by tierrif**
- [x] Setting to enable/disable this feature.

### CODE REVIEW

File: src/main/java/io/github/hotlava03/chatutils/fileio/ChatStorage.java [hunk 1]
```diff
@@ -25,32 +24,28 @@ public static ChatStorage getInstance() {
...
+    }
 
-        return StreamSupport.stream(arr.spliterator(), true)
-                .map(JsonElement::getAsString)
-                .toList();
+    public List<String> getStoredCmdLines(String server) {
+        return this.getLines("cmd.", server);
     }
 
     public void remove(String server, int index) {
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/fileio/ChatStorage.java:45]:
```suggestion
    public void removeChat(String server, int index) {
```

File: src/main/java/io/github/hotlava03/chatutils/mixin/JoinServerMixin.java [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+import java.util.List;
+
 @Mixin(ConnectScreen.class)
 public class JoinServerMixin {
+    @Mixin(ChatHud.class)
+    public interface MessageHistoryAccessor {
+        @Accessor
+        List<String> getMessageHistory();
+    }
+
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/mixin/JoinServerMixin.java]: Why is this here and not in its own mixin (use this mixin from the listener itself), JoinServerEvent has nothing to do with message history.

File: src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java [hunk 3]
```diff
@@ -2,39 +2,68 @@
...
-
-            lines.forEach((line) -> client.inGameHud.getChatHud().addMessage(Text.Serializer.fromJson(line)));
+        var client = MinecraftClient.getInstance();
+        var serverInfo = client.getCurrentServerEntry();
+        var address = serverInfo != null ? serverInfo.address : null;
+        if (address == null) {
+            LogManager.getLogger().error("Unexpected missing server info for joined server.");
             client.inGameHud.getChatHud().addMessage(Text.literal(
-                    StringUtils.translateAlternateColorCodes(Text.translatable("chat-utils.stored_messages", date).getString())
+                    StringUtils.translateAlternateColorCodes("&c[CHAT UTILS] &4Unexpected: Missing server/world info upon server/world join.")
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java:27]: is this gonna bug me whenever I join a single-player world

File: src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java [hunk 4]
```diff
@@ -2,39 +2,68 @@
...
+        chatLines.forEach((line) -> client.inGameHud.getChatHud().addMessage(Text.Serializer.fromJson(line)));
+        client.inGameHud.getChatHud().addMessage(Text.literal(
+                StringUtils.translateAlternateColorCodes(Text.translatable("chat-utils.stored_messages", date).getString())
+        ));
+    }
+
+    private void handleCommandPersist(ChatStorage storage, String address, List<String> messageHistory) {
+        var cmdLines = new ArrayList<>(storage.getStoredCmdLines(address));
+        if (cmdLines.isEmpty()) {
             storage.setBlockingChatEvents(false);
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java:65]: do chat events need to be blocked here, you're not modifying the HUD
just block and unblock in `handleChatPersist`
right and you're unblocking a 2nd time in the `accept` function anyway

File: src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java [hunk 5]
```diff
@@ -2,39 +2,68 @@
...
+            handleCommandPersist(storage, address, e.getMessageHistory());
+        }
+
+        storage.setBlockingChatEvents(false);
+    }
+
+    private void handleChatPersist(ChatStorage storage, String address, MinecraftClient client) {
+        var chatLines = new ArrayList<>(storage.getStoredChatLines(address));
+        if (chatLines.isEmpty()) {
+            storage.setBlockingChatEvents(false);
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java:51]: unnecessary to unblock here, but see below comment (blocking and unblocking should happen within this method itself and not in `accept`)

File: src/main/java/io/github/hotlava03/chatutils/mixin/JoinServerMixin.java [hunk 6]
```diff
@@ -9,6 +9,7 @@
 
 @Mixin(ConnectScreen.class)
 public class JoinServerMixin {
+
```
Comment (retrixe) [inline src/main/java/io/github/hotlava03/chatutils/mixin/JoinServerMixin.java:12]:
```suggestion
```

**Review by retrixe — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java (hunk 3 above)
Comment (tierrif) [reply src/main/java/io/github/hotlava03/chatutils/listeners/RetrieveChatListener.java:27]: No, ServerJoinEvent only works in multiplayer

**Review by retrixe — APPROVED** [review]
